tests/unit/test_calibration.py

```diff
@@ -123,14 +123,49 @@ def test_closest_point_fit() -> None:
     ), f"Offset is {linear_transform.offset} expected {expected_offset})"
 
 
+def test_closest_point_fit_gradient() -> None:
```

Comment:
note that the ggir functions should be renamed to indicate that they're ggir, as opposed to gradient desc.